twoliter/src/cmd/build.rs

```diff
@@ -234,7 +234,9 @@ mod test {
         );
 
         for package in packages {
-            let rpm = kit_output_dir.join(&format!("bottlerocket-{package}-0.0-0.{arch}.rpm"));
+            let rpm = kit_output_dir.join(&format!(
+                "bottlerocket-{package}-0.0-00000000000.00000000.br1.{arch}.rpm"
+            ));
```

Comment:
Will this test still succeed if the commit is `-dirty`?

Reply:
I'm confused as to how these default values are arriving in the `build-package` calls. I think this test is actually demonstrating the problem of having these args absent from `cargo make build-kit`. `build-kit` will result in package builds, and the package builds will then have these values absent.

I think buildsys needs to choke on the absence of these values and the Makefile needs to ensure they exist.

Reply:
> Will this test still succeed if the commit is `-dirty`?

If the commit were dirty, the rpm would be named `"bottlerocket-{package}-0.0-00000000000.00000000_dirty.br1.{arch}.rpm"` per https://github.com/bottlerocket-os/twoliter/pull/263/files#diff-a08098dd4b8cb6afda85dec0555534eaeb5a75f4e43f2db7ddcbc2e8c63d0e17R170-R171. So no, this test would not succeed if the package is being built from a dirty tree.

> I'm confused as to how these default values are arriving in the build-package calls. I think this test is actually demonstrating the problem of having these args absent from cargo make build-kit. build-kit will result in package builds, and the package builds will then have these values absent.

These values are set via environment variables in twoliter's Makefile.toml: https://github.com/bottlerocket-os/twoliter/pull/263/files#diff-ad5b91511e741abc0f447ce47f0d0c399ec15a4acd7666aa0f30094356d53af5R27-R31. They are read in `BuildPackageArgs` here: https://github.com/bottlerocket-os/twoliter/pull/263/files#diff-d69207eb08549801c5be3079f2fef42b9c132b83ecdb9d73160b91db074395cdR119-R135, which ultimately is used in a `build-package` call. I see between this and other similar comments, we must set these values in kit build as well, otherwise we'll always have default values present for packages built from kits...

> I think buildsys needs to choke on the absence of these values and the Makefile needs to ensure they exist.

can do

Reply:
I did some validation with kit building given these changes. Because kit building also builds packages, the required environment variables will be set for package building, and resultant kits will have packages with these values properly set:

```
$ ls tests/projects/local-kit/build/kits/extra-3-kit/aarch64/Packages
bottlerocket-pkg-e-0.0-01717804500.7185d90.br1.aarch64.rpm
bottlerocket-pkg-f-0.0-01717804500.7185d90.br1.aarch64.rpm
bottlerocket-pkg-g-0.0-01717804500.7185d90.br1.aarch64.rpm

$ ls tests/projects/local-kit/build/rpms/pkg-b
bottlerocket-pkg-b-0.0-01717804500.7185d90.br1.aarch64.rpm
```

As for why the integration test is using default values, that's because the integ test builds kits in a temp dir that and is not a git project.